ivadomed/loader/loader.py

```diff
@@ -424,14 +425,112 @@ def get_pair_slice(self, slice_index, gt_type="segmentation"):
 
         return dreturn
 
+    def read_file(self, filename):
+        """Read file according to file type.
+        Args:
+            filename (str): Subject filename.
+        """
+        if "nii" in self.extension:
+            # Returns 'nibabel.nifti1.Nifti1Image' object
+            # For 'nii' and 'nii.gz' extentions
+            return nib.load(filename)
+
+        elif self.extension == "png":
+            # Returns data from file as a 3D numpy array
+            # Behavior for grayscale png only, behavior TBD for RGB or RBGA png
+            return np.expand_dims(imageio.imread(filename, as_gray=True), axis=-1)
+
+        else:
+            raise RuntimeError("Input file type '{}' not supported".format(self.extension))
+
+    def apply_canonical(self, data):
+        """Apply nibabel as_closest_canonical function to nifti data only.
+        Args:
+            data ('nibabel.nifti1.Nifti1Image' object or 'ndarray'):
+                for nifti or png file respectively.
+        """
+        if "nii" in self.extension:
+            # Returns 'nibabel.nifti1.Nifti1Image' object
+            return nib.as_closest_canonical(data)
+
+        elif self.extension == "png":
+            # Returns data as is in numpy array
+            return data
+
+    def get_shape(self, data):
+        """Returns data shape according to file type.
+        Args:
+            data ('nibabel.nifti1.Nifti1Image' object or 'ndarray'):
+                for nifti or png file respectively.
+        Returns:
+            ndarray: Data shape.
+        """
+        if "nii" in self.extension:
+            return data.header.get_data_shape()
+
+        elif self.extension == "png":
+            return data.shape
+
+    def get_voxel_size(self, data):
+        """Returns voxel sizes in mm according to file type.
+        Args:
+            data ('nibabel.nifti1.Nifti1Image' object or 'ndarray'):
+                for nifti and png file respectively.
+        Returns:
+            tuple: Voxel size in mm
+        """
+        if "nii" in self.extension:
+            # Read zooms metadata from nifti file header
+            return data.header.get_zooms()
+
+        elif self.extension == "png":
+            # Voxel size for PNG is extracted from PixelSize metadata (from BIDS JSON sidecar)
+            # PixelSize definition in example dataset is a scalar in micrometers (BIDS BEP031 v 0.0.2)
+            # PixelSize definition may change for 2D [X, Y] and 3D [X, Y, Z] arrays in micrometers (BIDS BEP031 v 0.0.3)
+            # This method supports both behaviors.
+            # TODO: Update behavior to follow BEP microscopy development
+            # TODO: Review beahvior when implementing inference with PNG files (no BIDS metadata)
+            if 'PixelSize' in self.metadata[0]:
+                ps_in_um = self.metadata[0]['PixelSize']
+                if isinstance(ps_in_um, list) and (len(ps_in_um) == 2 or len(ps_in_um) == 3):
```

Comment:
For readability I would suggest to store numerical values in a variable. It would help the maintainer understand why we are checking for in this if statement

Reply:
Fixed in dfcdee2